pageserver/src/tenant.rs

```diff
@@ -1230,12 +1309,54 @@ impl Tenant {
                     )
                 }
                 Err(DownloadError::NotFound) => {
-                    info!("no index file was found on the remote");
+                    info!("no index file was found on the remote, found_delete_mark: {found_delete_mark}");
+
+                    if found_delete_mark {
+                        // We could've resumed at a point where remote index was deleted, but metadata file wasnt.
+                        // Cleanup:
+                        return DeleteTimelineFlow::cleanup_remaining_fs_traces_after_timeline_deletion(self, timeline_id)
+                            .await
+                            .context("cleanup_remaining_fs_traces_after_timeline_deletion")
+                            .map_err(LoadLocalTimelineError::resume_delete);
+                    }
+
+                    // We're loading fresh timeline that didnt yet make it into remote.
                     (None, Some(remote_client))
                 }
-                Err(e) => return Err(anyhow::anyhow!(e)),
+                Err(e) => {
+                    return Err(LoadLocalTimelineError::FailedToLoad {
+                        source: anyhow::anyhow!(e),
+                    })
+                }
             },
-            None => (None, remote_client),
+            None => {
+                // No remote client
+                if found_delete_mark {
+                    // There is no remote client, we found local metadata.
+                    // Continue cleaning up local disk.
+                    DeleteTimelineFlow::resume_deletion(
+                        Arc::clone(self),
+                        timeline_id,
+                        &local_metadata,
+                        None,
+                        init_order,
+                    )
+                    .context("resume deletion")
```

Comment:
This smells a bit like you are working around the `#[source]` I commented earlier, but here also the `LoadLocalTimelineError::resume_delete` is the context. Or maybe not, the `<Result<_, _> as Context>::context` is also used for `map_err(anyhow::Error::new)`.. Unsure.

Reply:
If feels redundant, I agree. Note that above there is this thing: 

```rust
self.create_timeline_delete_mark(timeline_id)
                                    .context("create delete mark")
                                    .map_err(LoadLocalTimelineError::resume_delete)?;
```

I still feel that context is useful here because it allows to know which one of the two calls lead to resume_delete error. If we had different error types there instead of anyhow this wouldnt've been needed. Do you agree?

Reply:
Aah and resume_delete must receive multiple different types as well. Yeah, context and the separation are useful. I'll resolve this as well.